### ISSUE TITLE
Revised Popups SOW - Alert  Boxes

### ISSUE THREAD

**Issue by FWAJL** [FWAJL/FieldWorkAssistantMVC#592]
All the details here: 
https://docs.google.com/document/d/1Ai74kFFUw_m02AhOb32gIRugvQHLGscmsGCs_iv8Nys/edit

**Comment by gsouvik** [comment]
@bsaiken @FWAJL 
I have a question regarding the requirement described in item 29 of Revised_Popups_SOW_3-4-2015.pdf. Is it required that the legend information would be editable in the xml? or it would be fixed for the this particular "type" of alerts?

Thank You.

**Comment by bsaiken** [comment]
@gsouvik 
I do not want the legend editable by the user.  I will need to build the legend contents.  Let me know how you want to do that.  Would you create a module that I could then fill in with html?  I will create the legend today so that you can use the contents as you see fit.

**Comment by gsouvik** [comment]
@bsaiken 
So basically each of those glyphicon­-question­-sign when clicked would show an alertbox which would have a number of legends displayed. I guess each such alertbox could have different set of legends. I think, as you have suggested, when you provide me the legend contents, I'll just put them into the system with "tags". And we can attach any number of legends with the alertbox just by those tags. This is how I plan to implement this feature.
Thank you.

**Comment by bsaiken** [comment]
@gsouvik 
There is only one legend.  I will give you a mockup later today and double check the implementation.

**Comment by bsaiken** [comment]
@gsouvik @FWAJL 
I have pushed changes to the feature_housekeeping branch with the map legend module (map_legend.php).  This is just for you to view, so I have temporarily placed it at the bottom of the Map > All Projects page (see image below).  You could hide the div and use jquery to view it with the legend glyphicon is clicked.

You will probably want to change the placement from the loadView.php module to the map.php module as it will be the same for every page.  

This legend needs some formatting fixes, and the contents are likely to change, but this should be enough to get you started.

I am guessing you will want to adapt my code and submit it under your feature branch.  Work out with Jeremie if commit https://github.com/FWAJL/FieldWorkAssistantMVC/commit/c3f14c91cf626bd7bc6467536245c72236404fba is to be rejected.  

I accidentally included a dummy ActiveTaskController file, which of course is not related to this Issue. 

![screen shot 2015-03-05 at 2 12 38 pm](https://cloud.githubusercontent.com/assets/8228624/6514724/cdc5249e-c341-11e4-95bf-f8f08a18c779.png)

**Comment by gsouvik** [comment]
@bsaiken @FWAJL 
Finished coding for the "Revised Popups SOW - Alert Boxes" and pushed to "feature_popup" branch.
1) Please review the legend alert box for map/* screens. I have just used the HTML you provided, put it into a module, and called it from the click event of the icon. 
2) For the Analyte section (item 30), I assumed if there is no field and lab analyte for the current project, the alert shows up. So basically if the left hand list is empty the alert fires. Please correct me if I was wrong and I shall change the implementation.
Thank you.

**Comment by bsaiken** [comment]
@FWAJL @gsouvik 
Looks great.

**Comment by FWAJL** [comment]
@gsouvik @bsaiken 

Here is my review:
1. Please put the text of the "OK" button in the resource file (line 91 in popup_maplegends).

I will merge to dev branch now and will merge the tiny update on Monday. Thank you.

**Comment by gsouvik** [comment]
@FWAJL @bsaiken 
I missed the 'OK' button earlier, added it in the resource file now.
Thank you.

**Comment by bsaiken** [comment]
@gsouvik 
New Item: 30a
on: click()    Analyte > List All
if : contents of: field-analyte-list and lab-analyte-list is empty
message: “You do not have any Analytes for your Projects.  Click "OK" to building your own custom analyte list.”
redirect: Clicking OK redirects to analyte/uploadList

Comment out item 30.  It is not what I intended.  Item 30a is what I intended.
Don’t delete the Alert box for item 30 - I may see the need for it later.
